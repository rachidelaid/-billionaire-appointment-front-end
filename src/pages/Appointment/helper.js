const handleErrors = (response) => {
  if (!response.ok) throw Error(response.statusText);
  return response;
};

export default handleErrors;

export const billionaires = [
  {
    id: 1,
    name: 'Elon Musk',
    title: 'CEO at SpaceX and Tesla, Inc.',
    image: 'https://github.com/orozCoding/billionares-pictures/blob/main/round_pictures/pc_elon.png?raw=true',
    price: 3250.0,
    description: 'Elon Musk is a business magnate and investor. He is the founder, CEO, and Chief Engineer at SpaceX; angel investor, CEO, and Product Architect of Tesla, Inc.; founder of The Boring Company; and co-founder of Neuralink and OpenAI. With an estimated net worth of around US$265 billion as of May 2022, Musk is the wealthiest person in the world according to both the Bloomberg Billionaires Index and the Forbes real-time billionaires list.',
    created_at: '2022-06-02T10:43:39.747Z',
    updated_at: '2022-06-02T10:43:39.747Z',
  },
  {
    id: 2,
    name: 'Jeff Bezos',
    title: 'Founder of Amazon',
    image: 'https://github.com/orozCoding/billionares-pictures/blob/main/round_pictures/pc_jeff.png?raw=true',
    price: 2650.0,
    description: "Jeff Bezos is an American entrepreneur, media proprietor, investor, computer engineer, and commercial astronaut. He is the founder, executive chairman and former president and CEO of Amazon. With a net worth of around US$140 billion as of May 2022, Bezos is the second-wealthiest person in the world and was the wealthiest from 2017 to 2021 according to both Bloomberg's Billionaires Index and Forbes.",
    created_at: '2022-06-02T10:43:39.762Z',
    updated_at: '2022-06-02T10:43:39.762Z',
  },
  {
    id: 3,
    name: 'Ariel Camus',
    title: 'Founder & CEO at Microverse',
    image: 'https://github.com/orozCoding/billionares-pictures/blob/main/round_pictures/pc_ariel.png?raw=true',
    price: 4350.0,
    description: 'The Argentine-born has founded multiple projects, including the one where is currently CEO: Microverse, a school that enables job-opportunities for talent around the world. Ariel has lived and worked in 7 different countries including Spain, Argentina, USA, and Vietnam and has been coding since he was 12. He previously founded Tourist Eye, a travel app with 1MM+ users acquired by Lonely Planet.',
    created_at: '2022-06-02T10:43:39.769Z',
    updated_at: '2022-06-02T10:43:39.769Z',
  },
  {
    id: 4,
    name: 'Julia Koch',
    title: 'President of David H. Koch Foundation',
    image: 'https://raw.githubusercontent.com/orozCoding/billionares-pictures/main/round_pictures/pc_julia.png',
    price: 2900.0,
    description: "Julia Koch is an American socialite, billionaire, philanthropist, and one of the richest women in the world. She inherited her fortune from her husband, David Koch, who died in 2019. As of January 2020, Forbes estimates her and her family's net worth at $43 billion, while Bloomberg estimates it at $61.6 billion.",
    created_at: '2022-06-02T10:43:39.777Z',
    updated_at: '2022-06-02T10:43:39.777Z',
  },
  {
    id: 5,
    name: 'Alice Walton',
    title: 'Heiress, Walton family fortune',
    image: 'https://raw.githubusercontent.com/orozCoding/billionares-pictures/main/round_pictures/pc_alice.png',
    price: 2800.0,
    description: 'Alice Walton is an American heiress to the fortune of Walmart. In September 2016, she owned over US$11 billion in Walmart shares.[3] As of March 2022, Walton has a net worth of $60.5 billion, making her the 20th-richest person, and the second richest woman in the world according to Bloomberg Billionaires Index.',
    created_at: '2022-06-02T10:43:39.784Z',
    updated_at: '2022-06-02T10:43:39.784Z',
  },
  {
    id: 6,
    name: 'Jack Ma',
    title: 'Co-founder of Yunfeng Capital',
    image: 'https://raw.githubusercontent.com/orozCoding/billionares-pictures/main/round_pictures/pc_jack.png',
    price: 2290.0,
    description: 'Jack Ma is a Chinese business magnate, investor and philanthropist. He is the co-founder and former executive chairman of Alibaba Group, a multinational technology conglomerate. In addition, he co-founded Yunfeng Capital, a Chinese private equity firm. Ma is a strong proponent of an open and market-driven economy.',
    created_at: '2022-06-02T10:43:39.792Z',
    updated_at: '2022-06-02T10:43:39.792Z',
  },
  {
    id: 7,
    name: 'Francoise Meyers',
    title: "Chairwoman at L'Oréal",
    image: 'https://github.com/orozCoding/billionares-pictures/blob/main/round_pictures/pc_francoise.png?raw=true',
    price: 3900.0,
    description: "Francoise Meyers is a French businesswoman, philanthropist, writer, pianist and billionaire heiress, the richest woman in the world, with an estimated net worth of US$75.3 billion as of March 2022, according to Forbes. She is the only child, heiress of Liliane Bettencourt and granddaughter of L'Oréal founder Eugène Schueller. Her mother died in September 2017, after which her fortune tripled with her investments through her family holding company, Tethys Invest, and the high valuation of L'Oréal shares on the stock exchange.",
    created_at: '2022-06-02T10:43:39.799Z',
    updated_at: '2022-06-02T10:43:39.799Z',
  },
  {
    id: 8,
    name: 'Changpeng Zhao',
    title: 'Founder & CEO at Binance',
    image: 'https://github.com/orozCoding/billionares-pictures/blob/main/round_pictures/pc_chang.png?raw=true',
    price: 3500.0,
    description: "Changpeng Zhao commonly known as 'CZ', is a Chinese-Canadian business executive. Zhao is the founder and CEO of Binance, the world's largest cryptocurrency exchange by trading volume as of April 2018. CZ was born in Jiangsu, China and moved to Canada at a young age with his family.",
    created_at: '2022-06-02T10:43:39.806Z',
    updated_at: '2022-06-02T10:43:39.806Z',
  },
  {
    id: 9,
    name: 'Mark Zuckerberg',
    title: 'Co-founder & CEO of Meta Platforms',
    image: 'https://github.com/orozCoding/billionares-pictures/blob/main/round_pictures/pc_mark.png?raw=true',
    price: 2980.0,
    description: 'Mark Zuckerberg is an American media magnate, internet entrepreneur, and philanthropist. He is known for co-founding the social media website Facebook and its parent company Meta Platforms (formerly Facebook, Inc.), of which he is the chairman, chief executive officer, and controlling shareholder.',
    created_at: '2022-06-02T10:43:39.813Z',
    updated_at: '2022-06-02T10:43:39.813Z',
  },
  {
    id: 10,
    name: 'Bill Gates',
    title: 'Co-founder of Microsoft',
    image: 'https://github.com/orozCoding/billionares-pictures/blob/main/round_pictures/pc_bill.png?raw=true',
    price: 3120.0,
    description: 'Billgates is an American business magnate, software developer, investor, author, and philanthropist. He is a co-founder of Microsoft, along with his late childhood friend Paul Allen. During his career at Microsoft, Gates held the positions of chairman, chief executive officer (CEO), president and chief software architect, while also being the largest individual shareholder until May 2014. He was a major entrepreneur of the microcomputer revolution of the 1970s and 1980s.',
    created_at: '2022-06-02T10:43:39.819Z',
    updated_at: '2022-06-02T10:43:39.819Z',
  },
];
