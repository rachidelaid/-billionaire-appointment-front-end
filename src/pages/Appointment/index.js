import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import style from './style.module.css';
import handleErrors from './helper';

const Appointment = () => {
  const [city, setCity] = useState('');
  const [date, setDate] = useState('');
  const [billionaire_id, setBillionaire_id] = useState('');
  const navigate = useNavigate();

  const createAppointment = (e) => {
    e.preventDefault();

    const appointment = {
      city, date, billionaire_id, user_id: 1,
    };

    fetch('http://localhost:3000/api/appointments', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ appointment }),
    }).then(handleErrors)
      .then(() => navigate('/'))
      .catch((error) => console.log(error));
  };

  return (
    <div className={`${style.container} ${style['flex-center']}`}>
      <div className={`${style.information} ${style['flex-center']}`}>
        <h1 className={style.heading}>BOOK A BILLIONAIRE</h1>
        <hr className={style.line} />
        <p className={style.description}>
          Ever wanted to spend a day with a billionaire genius who can give you advice
          on how to run a business?
          Maybe you just wanted a coffee-chat with him to see the personality behind
          the money. Well, in Billionaire Appointment this is possible. If you wish
          to book YOUR billionaire, please use the selector below.
        </p>
        <form className={style.form} onSubmit={(e) => createAppointment(e)}>
          <input
            className={style['form-child']}
            type="text"
            placeholder="City"
            value={city}
            onChange={(e) => setCity(e.target.value)}
            required
          />
          <select
            className={style['form-child']}
            value={billionaire_id}
            onChange={(e) => setBillionaire_id(e.target.value)}
            required
          >
            <option value="" disabled>
              Billionaires List
            </option>
            <option value="1">
              Ariel Camus
            </option>
            <option value="2">
              Bill Gates
            </option>
            <option value="3">
              Francoise Meyers
            </option>
            <option value="4">
              Jack Ma
            </option>
          </select>
          <input
            className={style['form-child']}
            type="date"
            value={date}
            onChange={(e) => setDate(e.target.value)}
            required
          />
          <button
            className={style.btn}
            type="submit"
          >
            Book Now
          </button>
        </form>
      </div>
    </div>
  );
};

export default Appointment;
