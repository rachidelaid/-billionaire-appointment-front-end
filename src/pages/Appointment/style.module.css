.flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.container {
  padding: 1rem;
  background: url("https://www.zastavki.com/pictures/1920x1080/2013/Aviation_A_small_private_plane_in_flight_043452_23.jpg") no-repeat center, var(--green);
  background-size: cover;
  height: 100vh;
  background-blend-mode: soft-light;
  flex-direction: column;
}

.information {
  flex-direction: column;
}

.heading,
.description {
  color: #fff;
}

.heading {
  font-family: var(--font-bold);
  letter-spacing: 0.5rem;
  text-align: center;
}

.line {
  width: 40%;
  margin: 1rem;
  border: 1px solid rgba(255, 255, 255, 0.7);
}

.description {
  width: 65%;
  margin-bottom: 1rem;
}

.login-required {
  display: flex;
  flex-direction: column;
  align-items: center;
  color: #fff;
}

@media only screen and (max-width: 670px) {
  .line,
  .description {
    width: 90%;
  }

  .container {
    background-size: cover;
    background-position: 30%;
  }
}
