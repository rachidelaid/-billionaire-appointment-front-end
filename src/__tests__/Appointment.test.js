import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Appointment from '../pages/Appointment';

const RouterProvider = () => (
  <BrowserRouter>
    <Appointment />
  </BrowserRouter>
);

describe('Appointments Component', () => {
  it('renders', () => {
    render(<RouterProvider />);
    const welcomeText = screen.getByText('BOOK A BILLIONAIRE');
    const cityInput = screen.getByPlaceholderText('City');
    const billionaireSelect = screen.getByTestId('select-tag');
    const dateInput = screen.getByTestId('date-input');
    const submitButton = screen.getByText('Book Now');

    expect(welcomeText).toMatchSnapshot();
    expect(cityInput).toMatchSnapshot();
    expect(billionaireSelect).toMatchSnapshot();
    expect(dateInput).toMatchSnapshot();
    expect(submitButton).toMatchSnapshot();
  });
});
