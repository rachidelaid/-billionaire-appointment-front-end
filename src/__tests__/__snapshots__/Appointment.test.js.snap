// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments Component should create an appointment 1`] = `Promise {}`;

exports[`Appointments Component should not create an appointment 1`] = `
<button
  class="btn"
  type="submit"
>
  Book Now
</button>
`;

exports[`Appointments Component should render input date 1`] = `
<input
  class="form-child"
  placeholder="Select date"
  required=""
  type="date"
  value=""
/>
`;

exports[`Appointments Component should render input text 1`] = `
<input
  class="form-child"
  placeholder="City"
  required=""
  type="text"
  value=""
/>
`;

exports[`Appointments Component should render select tag 1`] = `
<select
  class="form-child"
  required=""
>
  <option
    disabled=""
    value=""
  >
    Billionaires List
  </option>
  <option
    value="1"
  >
    Ariel Camus
  </option>
  <option
    value="2"
  >
    Bill Gates
  </option>
  <option
    value="3"
  >
    Francoise Meyers
  </option>
  <option
    value="4"
  >
    Jack Ma
  </option>
</select>
`;

exports[`Appointments Component should render submit button 1`] = `
<button
  class="btn"
  type="submit"
>
  Book Now
</button>
`;

exports[`Appointments Component should render welcome text 1`] = `
<h1
  class="heading"
>
  BOOK A BILLIONAIRE
</h1>
`;
